ENT-9083: Updated docs to include new behaviour of package[update]smatching

larsewi opened the pull request.

larsewi committed:
Updated docs to include new behaviour of package[update]smatching

Ticket: ENT-9083
Changelog: None
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>

craigcomstock commented on reference/functions/packagesmatching.markdown:
```diff
@@ -42,8 +42,10 @@ some desired packages, and finally reports if they are installed.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions silmontaneously.
```
```suggestion
At no time will both the standard and the legacy data be available to these functions simultaneously.
```

craigcomstock commented on reference/functions/packageupdatesmatching.markdown:
```diff
@@ -39,8 +39,10 @@ this:
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions silmontaneously.
```
```suggestion
At no time will both the standard and the legacy data be available to these functions simultaneously.
```

larsewi committed:
Apply typo correction suggestions from Craig

Co-authored-by: Craig Comstock <craig.comstock@northern.tech>

nickanderson commented on reference/functions/packagesmatching.markdown:
```diff
@@ -42,8 +42,10 @@ some desired packages, and finally reports if they are installed.
 Since only one `body common control` can be present in a policy set any bundles which use these functions will typically need to execute in the context of a full policy run.
-If there is no `package_inventory` attribute such as on package module unsupported platforms or when a policy entry file other than promises.cf is selected with the `--file -f` argument then the legacy package methods data will be used.
-At no time will both standard and legacy data be available to these functions.
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
```
3.22? Won't this be 3.23?

And then if it's backported, 3.18.6, 3.21.3

nickanderson replied:
Also seems liek this should be mentioned in the history section.

nickanderson commented on reference/functions/packageupdatesmatching.markdown:
```diff
@@ -39,8 +39,10 @@ this:
+However, since the release of CFEngine 3.22, the `packagesmatching` and `packageupdatesmatching` policy functions will look for and use the existing software inventory databases (available in `$(sys.statedir)`), if the default package inventory is not configured.
+This change enables the usage of these policy functions in standalone policy files. But please note that you still need the default package inventory attribute specified in the policy framework for the software inventory databases to exist in the first place and for them to be maintained/updated.
+If there is no `package_inventory` attribute (such as on package module unsupported platforms) and there are no software inventory databases available in `$(sys.statedir)` then the legacy package methods data will be used instead.
+At no time will both the standard and the legacy data be available to these functions simultaneously.
```
Seems like this should be noted in the History section

nickanderson committed:
Tidied up package[update]smatching() docs

- Removed Argument Descriptions (it's duplicating the function attributes macro)
- Moved historical reference to history section

nickanderson commented:
@cf-bottom jenkins, please

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=fast-build-and-deploy-docs-master&build=376)](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/376/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/376/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-376/output/_site/

nickanderson committed:
Tidy inline example for packageupdatesmatching()

cf-bottom commented:
Sure, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=fast-build-and-deploy-docs-master&build=377)](https://ci.cfengine.com//job/fast-build-and-deploy-docs-master/377/)

**Jenkins:** https://ci.cfengine.com/job/fast-build-and-deploy-docs-master/377/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-fast-build-and-deploy-docs-master-377/output/_site/

nickanderson committed:
Adjusted vertical alignment of arguments for rendered example

Strangely, alignment in text does not mean alignment in the rendered version.

nickanderson commented:
@cf-bottom jenkins, please

larsewi commented:
Cherry picked to
 - https://github.com/cfengine/documentation/pull/3073
 - https://github.com/cfengine/documentation/pull/3074
